doc/api/errors.md

```diff
@@ -1073,6 +1073,11 @@ and `:method`) may be used.
 An action was performed on an `Http2Session` object that had already been
 destroyed.
 
+<a id="ERR_HTTP2_INVALID_SETTING_OBJECT"></a>
+### `ERR_HTTP2_INVALID_SETTING_OBJECT`
```

Comment:
@lundibundi Is this the kind of error code you were looking to reduce/eliminate with c66e6471e780e752f3f6f9337ac49a3d6e3d7091? Should this change use a more general code like `ERR_INVALID_ARG_VALUE`? Or is this appropriate in this case?

Reply:
I'd say yes since this is a basic synchronous argument validation it should just use `ERR_INVALID_ARG_VALUE`.

Reply:
After checking this in the http2, the settings validation is handled via `assertIsObject(settings, 'settings');` which uses `ERR_INVALID_ARG_TYPE` so we should probably be consistent here and just use it for now.